scalafmt-core/shared/src/main/scala/org/scalafmt/config/ScalafmtRunner.scala

```diff
@@ -88,6 +88,10 @@ object ScalafmtRunner {
       .withAllowTrailingCommas(true)
     val default = scala213
     val Scala3 = Dotty
+      // Disable significant indentation because the exact syntax is still under
+      // development and is likely to change soon. See discussions at
+      // https://contributors.scala-lang.org/t/feedback-sought-optional-braces/4702
+      .withAllowSignificantIndentation(false)
 
```

Comment:
not a big fan of latent changes.

Reply:
Unfortunately, I think until it all is stable we shouldn't enable it in scalafmt.

Reply:
I created a separate PR for some of the unrelated changes in this PR: https://github.com/scalameta/scalafmt/pull/2312

Reply:
I will rebase once the change is made.